cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -1078,4 +1080,19 @@ public static String clusterCaKeySecretName(String cluster)  {
 
         return merged;
     }
+
+    public ServiceAccount generateServiceAccountInternal() {
```

Comment:
Why do we have these two methods? Why is this not part of the `generateServiceAccount` directly?

Reply:
In `EntityOperator` we perform `isDeployed` check so we need to use this like that.

Reply:
Ok, lets see how @tombentley likes it ... maybe he proposes a better solution :-)

Reply:
I think in EO you can just override

    @Override
    public ServiceAccount generateServiceAccount() {

        if (!isDeployed()) {
            return null;
        }
        return super.generateServiceAccount()
    }

So there's no need for the internal method.